subsys/net_core_monitor/app_core.c

```diff
@@ -17,7 +20,12 @@ LOG_MODULE_REGISTER(net_core_monitor, CONFIG_NET_CORE_MONITOR_LOG_LEVEL);
 #define IPC_MEM_CNT_IDX  0
 #define IPC_MEM_REAS_IDX 1
 
-int ncm_net_status_check(uint32_t * const reset_reas)
+#define NET_CORE_CHECK_INTERVAL_MSEC (CONFIG_NCM_FEEDING_INTERVAL_MSEC + 500)
```

Comment:
I think this is misleading. `CONFIG_NCM_FEEDING_INTERVAL_MSEC` affects power consumption on both cores so we shouldn't hardcode any offset here.

My suggestion is to change the definition of the symbol in Kconfig so that is has different default depending on the core. Also, the requirement for the interval on the application core being not lesser than the internal on the network core should be documented. Something like:
```
config NCM_FEEDING_INTERVAL_MSEC
	default 500 if SOC_NRF5340_CPUNET
        default 1000 if SOC_NRF5340_CPUAPP
	int "Feeding interval in milliseconds"
        help
          The value of this parameter on the application core must be greater
          than the value on the network core. Otherwise the network core monitor
          will report false positive network code malfunctions.
```

or similar.

Reply:
Good point. I have applied your suggestions.